packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -328,6 +334,12 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
                 valueExpression = Literal((inputParam.Type as InputLiteralType)?.Value);
                 format = ClientModelPlugin.Instance.TypeFactory.GetSerializationFormat(inputParam.Type).ToFormatSpecifier();
             }
+            else if (TryGetSpecialHeaderParam(inputParam, out var parameterProvider) &&
+                parameterProvider.DefaultValue != null)
```

Comment:
nit: since we are constructing these as static values are these default values ever going to be null?